utils/buffer/src/slice.rs

```diff
@@ -135,25 +269,66 @@ impl From<Vec<u8>> for Slice {
     }
 }
 
+// The shared state of the buffer pool.
+//
+// Encapsulates an atomic 8-bit value (`AtomicU8`) to track the shared state of memory slices in a
+// thread-safe manner. It uses atomic operations to ensure that memory tracking can be done
+// concurrently without locks.
+//
+// Each bit in the `AtomicU8` represents the state of a memory slot (e.g., whether it is allocated
+// or free) in the buffer pool, allowing the system to manage and synchronize memory usage across
+// multiple slices.
+//
+// `SharedState` acts like a reference counter, helping the buffer pool know when a buffer slice is
+// safe to clear. Each time a memory slice is used or released, the corresponding bit in the shared
+// state is toggled. When no slices are in use (all bits are zero), the buffer pool can safely
+// reclaim or reuse the memory.
+//
+// This system ensures that no memory is prematurely cleared while it is still being referenced.
+// The buffer pool checks whether any slice is still in use before clearing, and only when the
+// shared state indicates that all references have been dropped (i.e., no unprocessed messages
+// remain) can the buffer pool safely clear or reuse the memory.
 #[derive(Clone, Debug)]
 pub struct SharedState(Arc<AtomicU8>);
 
 impl Default for SharedState {
+    // Creates a new `SharedState` with an internal `AtomicU8` initialized to `0`, indicating no
+    // memory slots are in use.
     fn default() -> Self {
         Self::new()
     }
 }
 
 impl SharedState {
+    // Creates a new `SharedState` with an internal `AtomicU8` initialized to `0`, indicating no
+    // memory slots are in use.
```

Comment:
Why did you use  normal comments and not public docs ones here?

Reply:
It's private.